### DISCUSSION TITLE
Set periodic boundary to calculate the rotationally symmetric structure

### DISCUSSION THREAD

**Discussion by lucuo** [idaholab/moose#27241] [Q&A Modules: Solid mechanics]
Hello, I want to calculate a quarter ring using a rotationally symmetric periodic boundary, using a [periodic] boundary set on both sides of the figure below, and set a symmetric constraint on one side. But the mechanical calculation was wrong, it seems, because moose's periodic boundary simply equates the values of disp_x and disp_y at the two boundaries, whereas I wanted their displacement vectors to be equal after the Angle transformation.
 Is there any other way to set periodic boundaries of rotational symmetry (besides setting symmetric constraints on both sides)?
![屏幕截图 2024-03-30 121646](https://github.com/idaholab/moose/assets/114867841/759a3b03-72c2-4297-a6ec-459c439db14b)
![图片2](https://github.com/idaholab/moose/assets/114867841/b3265e0e-a652-49a9-bd85-084548f8b9f4)

**Comment by GiudGiud** [comment]
Hello

If you are not going to use MPI / distributed paralellism, this can be coded up fairly easily using functors. Functors can be evaluated arbitrarily anywhere on the domain, so on the left boundary you could evaluate the functors (here variables) on the bottom boundary
This post did this for some sort of shifted periodic boundary condition:
https://github.com/idaholab/moose/discussions/26854

Guillaume

**Reply by lucuo** [reply]
Thank you for your answer. But my problems don't seem to be quite the same as his. 
My two-dimensional model divides the displacement vector into disp_x and disp_y in cartesian coordinates, and  I want to equal the  displacement vector through the angle transformation on both boundaries.
![屏幕截图 2024-03-30 233653](https://github.com/idaholab/moose/assets/114867841/78aaad67-d3f9-4806-9450-1dcc1649c5ab)
Similar to the cyclic symmetric periodic boundary in COMSOL.

**Reply by GiudGiud** [reply]
It s not the same so you can’t use their code. 

But it s very similar, you could solve this with a dirichlet BC on one surface that is set to the desired transformation of the displacement on the other.

If you are asking if we have a cyclic symmetric periodic BC in moose the answer is not. We don’t have any immediate plans for it either. If you wanted to implement it, that would be a good thing

**Reply by lucuo** [reply]
If I want to write cyclic symmetric periodic boundaries. Is it possible to modify the [constraints] module？

**Reply by GiudGiud** [reply]
That's a good idea as well.
